python/paddle/v2/fluid/framework.py

```diff
@@ -695,11 +702,22 @@ def var_recursive(self, name):
             return self.var(name)
         else:
             if self.idx == 0:
-                raise ValueError("var %s is not in block(%d) nor its parents." %
-                                 name, self.idx)
+                raise ValueError(
+                    "var {0} is not in block({1}) nor its parents.".format(
+                        name, self.idx))
             else:
-                parent_block = self.program.block(self.parent_idx)
-                return parent_block.var_recursive(name)
+                # DFS
+                try:
+                    parent_block = self.program.block(self.parent_idx)
+                    return parent_block.var_recursive(name)
+                except ValueError:
+                    fwd_block = self.program.block(
+                        self.forward_block_idx
+                    ) if self.forward_block_idx != -1 else None
+                    if fwd_block is not None:
+                        return fwd_block.var_recursive(name)
+                    else:
+                        raise
```

Comment:
I am not familiar with the python syntax. What is being raised here?

Reply:
This statement will rethrow the exception.